.github/workflows/e2e-on-release.yml

```diff
@@ -0,0 +1,331 @@
+name: Run E2E Tests on Release PRs
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.event.pull_request.title, '(release)') }}
+    outputs:
+      patient_management_version: ${{steps.versions.outputs.patient_management}}
+      patient_chart_version: ${{steps.versions.outputs.patient_chart}}
+      esm_core_version: ${{steps.versions.outputs.esm_core}}
+      form_builder_version: ${{steps.versions.outputs.form_builder}}
+      cohort_builder_version: ${{steps.versions.outputs.cohort_builder}}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Checkout to the release commit
+        run: |
+          release_commit_sha=$(git log --oneline | awk 'NR==3 {print $1}')
+          git checkout $release_commit_sha
+
+      - name: Extract version numbers from the spa-build-config.json file
+        id: versions
+        run: sh e2e_test_support_files/extract_tag_numbers.sh
+
+      - name: Build and Run Containers
+        run: docker-compose -f e2e_test_support_files/docker-compose-build.yml up -d
+
+      - name: Wait for the backend to start
+        run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8080/openmrs/login.htm)" != "200" ]]; do sleep 10; done
+
+      - name: Commit and export Containers
+        run: sh e2e_test_support_files/commit_and_export_images.sh
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: e2e_release_env_images
+          path: e2e_release_env_images.tar.gz
+          retention-days: 1
+
+  run-patient-management-e2e-tests:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download Docker Images
+        uses: actions/download-artifact@v3
+        with:
+          name: e2e_release_env_images
+          path: /tmp
```

Comment:
It might be slightly messy, but it's a little better to have a step that uses `mktemp -d` and saves the output in a output variable, which is then used in-place of `/tmp`. This makes the job less dependent on GH Actions providing us with a clean environment.

Reply:
fixed with 29efd49